src/api/v2/v2AzureResourcesApi.ts

```diff
@@ -148,7 +148,7 @@ export interface V2AzureResourcesApi extends AzureResourcesApiBase {
     /**
      * Show a quick picker of app resources. Set `options.type` to filter the picks.
      */
-    pickResource<TModel>(options?: ResourcePickOptions): vscode.ProviderResult<TModel>
+    pickResource<TModel extends ResourceModelBase>(actionContext: IActionContext, options?: ResourcePickOptions): vscode.ProviderResult<TModel | ApplicationResource>;
```

Comment:
I was thinking that there would be an expectation that callers of `pickResource<TModel>()` would have specific knowledge of the returned model type.  That is, an extension exposing a command that picks one of its resources would always expect to get back one of its model types, even if the thing being picked was an application resource (i.e. otherwise an `ApplicationResource`).  Hence, returning `TModel | ApplicationResource` seems a little odd to me.  Even in the case where the Resource Groups extension is doing the picking, it'd still (I would think) expect to get back its own model type, which isn't an `ApplicationResource` but, rather, a type that wraps an `ApplicationResource`.